docs/source/advanced/validator.rst

```diff
@@ -0,0 +1,61 @@
+errors found.
+
+Valid Topologies
+++++++++++++++++
+
+.. code-block:: bash
+
+   $ linchpin validate
+   topology for target 'libvirt-new' is valid
+   topology for target 'libvirt' is valid under old schema
+   topology for target 'libvirt-network' is valid
+
+
+Invalid Topologies
+++++++++++++++++++
+
+This is an example of a topology which does not validate
+
+.. code-block:: bash
+
+   $ linchpin validate
+   topology for target 'libvirt-network' is valid
+
+   Topology for target 'libvirt-new' does not validate
+   topology: 'OrderedDict([('topology_name', 'libvirt-new'), ('resource_groups', [OrderedDict([('resource_group_name', 'libvirt-new'), ('resource_group_type', 'libvirt'), ('resource_definitions', [OrderedDict([('role', 'libvirt_node'), ('uri', 'qemu:///system'), ('image_src', 'http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz'), ('memory', 2048), ('vcpus', '1'), ('arch', 'x86_64'), ('ssh_key', 'libvirt'), ('networks', [OrderedDict([('name', 'default'), ('hello', 'world')])]), ('additional_storage', '10G'), ('cloud_config', OrderedDict([('users', [OrderedDict([('name', 'herlo'), ('gecos', 'Clint Savage'), ('groups', 'wheel'), ('sudo', 'ALL=(ALL) NOPASSWD:ALL'), ('ssh-import-id', 'gh:herlo'), ('lock_passwd', True)])])])), ('count', 1)])])])])])'
+   errors:
+         res_defs[0][count]: value for field 'count' must be of type 'integer'
+         res_defs[0][networks][0][hello]: field 'hello' could not be recognized within the schema provided
+         res_defs[0][name]: field 'name' is required
+
+
+The `linchpin validate` command can also provide a list of errors against the
+old schema with the `--old-schema` floag
+
+.. code-block:: bash
+
+   $ linchpin validate --old-schema
+   
+   Topology for target 'libvirt-new' does not validate
+   topology: 'OrderedDict([('topology_name', 'libvirt-new'), ('resource_groups', [OrderedDict([('resource_group_name', 'libvirt-new'), ('resource_group_type', 'libvirt'), ('resource_definitions', [OrderedDict([('role', 'libvirt_node'), ('uri', 'qemu:///system'), ('image_src', 'http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz'), ('memory', 2048), ('vcpus', '1'), ('arch', 'x86_64'), ('ssh_key', 'libvirt'), ('networks', [OrderedDict([('name', 'default'), ('hello', 'world')])]), ('additional_storage', '10G'), ('cloud_config', OrderedDict([('users', [OrderedDict([('name', 'herlo'), ('gecos', 'Clint Savage'), ('groups', 'wheel'), ('sudo', 'ALL=(ALL) NOPASSWD:ALL'), ('ssh-import-id', 'gh:herlo'), ('lock_passwd', True)])])])), ('count', 1)])])])])])'
+   errors:
+         res_defs[0][networks][0][hello]: field 'hello' could not be recognized within the schema provided
+         res_defs[0][name]: field 'name' is required
+
+As you can see, the error stating the 'count' must be an integer is gone, because the old schema makes no assumptions about the type of 'count'.
```

Comment:
In both cases above, maybe show a simple topology which is being validated? it's also hard to  know what is right and how to correct the schema. Maybe even add an update to show how the schema (just one) is corrected and passes the validator?